### PR TITLE
Upgrade deprecated react-component modals to USWDS v3 web-component modals

### PR DESCRIPTION
**Pull request by powellkerry**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Changed occurrences of @department-of-veterans-affairs/component-library/Modal to the USWDS v3 web-component equivalent 

## Related issue(s)
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2450

## Testing done
local testing, updated unit tests

## What areas of the site does it impact?
- medical-copays
- personalization

## Acceptance criteria
- Affected pages and modals still work correctly

### Quality Assurance & Testing

- [ x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ x] Linting warnings have been addressed
- [ x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ x] Screenshot of the developed feature is added
- [ x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ x] Browser console contains no warnings or errors.
- [ x] Events are being sent to the appropriate logging solution
- [ x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by OptionSelect — DISMISSED** [review]
LGTM! 🚀

**Comment by adamwhitlock1** [comment]
i actually have a PR already pending for the personalization work, I did just add a commit for the useDowntimeRenderMethod as well to update that modal, so that should cover our application code. maybe you can merge this with main once mine has merge. Sorry didn't realize you were doing this work as well.

https://github.com/department-of-veterans-affairs/vets-website/pull/27830

**Review by adamwhitlock1 — APPROVED** [review]

**Review by it-harrison — APPROVED** [review]
lgtm

**Pull request merged**
